source/include/core_json.h

```diff
@@ -84,18 +84,21 @@ JSONStatus_t JSON_Validate( const char * buf,
 /* @[declare_json_validate] */
 
 /**
- * @brief Find a key in a JSON object and output the pointer @p outValue
- * to its value.
+ * @brief Find a key or array index in a JSON document and output the
+ * pointer @p outValue to its value.
  *
- * The JSON document must contain an object (e.g., <code>{"key":"value"}</code>).
- * Any value may also be an object and so forth to a maximum depth.  A search
- * may descend through nested objects when the queryKey contains matching
- * key strings joined by a separator.
+ * Any value may also be an object or an array to a maximum depth.  A search
+ * may descend through nested objects or arrays when the queryKey contains matching
+ * key strings or array indexes joined by a separator.
  *
  * For example, if buf contains <code>{"foo":"abc","bar":{"foo":"xyz"}}</code>,
```

Comment:
Better to use a proper phrase for the buffer or refer to the input parameter as `@p buf`
```suggestion
 * For example, if the provided buffer contains <code>{"foo":"abc","bar":{"foo":"xyz"}}</code>,
```

Reply:
changed